tests/common/include/tests/common/path_correction.inl

```diff
@@ -7,463 +7,171 @@
-    EXPECT_NEAR(crk_state().pos()[2], 0.f, env::tol);    // z
-
-    // Bound state after one turn propagation
-    const auto bound_param1 = crk_state._bound_params;
-    const auto bound_vec0 = bound_param0.vector();
-    const auto bound_vec1 = bound_param1.vector();
-    const auto bound_cov0 = bound_param0.covariance();
-    const auto bound_cov1 = bound_param1.covariance();
-
-    // Check if the bound state stays the same after one turn propagation
-
-    // vector
-    for (unsigned int i = 0u; i < e_bound_size; i++) {
-        EXPECT_NEAR(matrix_operator().element(bound_vec0, i, 0u),
-                    matrix_operator().element(bound_vec1, i, 0u), env::tol);
+    getter::element(bound_cov0, e_bound_loc0, e_bound_loc0) = 1.f;
+    getter::element(bound_cov0, e_bound_loc1, e_bound_loc1) = 1.f;
+    getter::element(bound_cov0, e_bound_phi, e_bound_phi) = 1.f;
+    // Set theta error to zero, to suppress the loc1 (z) divergence
+    getter::element(bound_cov0, e_bound_theta, e_bound_theta) = 0.f;
+    getter::element(bound_cov0, e_bound_qoverp, e_bound_qoverp) = 1.f;
+    getter::element(bound_cov0, e_bound_time, e_bound_time) = 0.f;
+
+    // Copy the covariance
+    typename bound_track_parameters<transform3>::covariance_type bound_cov =
+        bound_cov0;
+
+    // Total path length, just for testing purpose
+    scalar total_path_length = 0.f;
+
+    // Iterate over the planes until we reach the first plane (one loop)
+    for (std::size_t i_p = 0u; i_p < n_planes; i_p++) {
+
+        // Next surface index (circular)
+        const std::size_t next_index = i_p == n_planes - 1 ? 0u : i_p + 1;
+
+        // Helix on the current surface
+        detail::helix<transform3> hlx(free_trk, &B);
+
+        // bound vector on the current surface
+        const auto bound_vec =
+            c2.free_to_bound_vector(trfs[i_p], free_trk.vector());
+
+        // Bound-to-free jacobian on the current surface
+        const typename cartesian2<transform3>::bound_to_free_matrix
+            bound_to_free_jacobi =
+                c2.bound_to_free_jacobian(trfs[i_p], rectangle, bound_vec);
+
+        // Get the intersection on the next surface
+        const auto is = hpi(hlx, rectangle, trfs[next_index])[0];
+
+        // Helical path length between two surfaces
+        const auto path_length = is.path;
+
+        // Position on the next surface
+        const auto new_pos = hlx(path_length);
+
+        // Direction on the next surface
+        const auto new_dir = hlx.dir(path_length);
+
+        // Add the path length to the total path length
+        total_path_length += path_length;
+
+        // Free transport jacobian between two surfaces
+        const cartesian2<transform3>::free_matrix transport_jacobi =
+            hlx.jacobian(path_length);
+
+        // Reset the free track parameters with the position and direction on
+        // the next surface
+        free_trk.set_pos(new_pos);
```

Comment:
Are the ```new_pos``` and ```new_dir``` variables really needed?

Reply:
Maybe not. I just put them into set_pos and set_dir function directly